### ISSUE TITLE
What does this line do?

### ISSUE THREAD

**Issue by ovshake** [MadryLab/mnist_challenge#12]
Hi, 
What does [this](https://github.com/MadryLab/mnist_challenge/blob/1756841ce47a766bc5a700db371eef3eebac5d2b/pgd_attack.py#L36) line of code do?  How was the constant of `50` decided? I am trying to reproduce this in PyTorch.

Many thanks

**Comment by dtsip** [comment]
This corresponds to the [Carlini-Wagner](https://arxiv.org/abs/1608.04644) loss function. You can see their paper for details and motivation. The constant is arbitrary, any constant >0 would work just as well for white-box attacks.
